codec-http2/src/main/java/io/netty/handler/codec/http2/HttpToHttp2ConnectionHandler.java

```diff
@@ -61,27 +68,92 @@ private int getStreamId(HttpHeaders httpHeaders) throws Exception {
+                                promiseAggregator.newPromise());
+                        release = false;
+
+                        if (hasTrailingHeaders) {
+                            encoder.writeHeaders(ctx, currentStreamId, trailingHeaders, 0, true,
+                                    promiseAggregator.newPromise());
+                        }
+
+                        promiseAggregator.doneAllocatingPromises();
+                    } else {
+                        encoder.writeHeaders(ctx, currentStreamId, http2Headers, 0, true, promise);
+                    }
+                } else {
+                    encoder.writeHeaders(ctx, currentStreamId, http2Headers, 0, false, promise);
+                }
+            } catch (Throwable t) {
+                if (promiseAggregator == null) {
+                    promise.tryFailure(t);
+                } else {
+                    promiseAggregator.setFailure(t);
+                }
+            } finally {
+                if (release) {
+                    ReferenceCountUtil.release(httpMsg);
+                }
+            }
+        } else if (msg instanceof HttpContent) {
+            final HttpContent httpContent = (HttpContent) msg;
+            boolean release = true;
+            SimpleChannelPromiseAggregator promiseAggregator = null;
+            try {
+                if (currentStreamId == 0) {
+                    throw new IllegalStateException("currentStreamId is zero");
+                }
+
+                Http2ConnectionEncoder encoder = encoder();
+                if (httpContent instanceof LastHttpContent) {
+                    // Convert and potentially write any trailing headers.
+                    final LastHttpContent lastContent = (LastHttpContent) httpContent;
+                    final Http2Headers http2Headers = lastContent.trailingHeaders().isEmpty() ? null :
+                            HttpUtil.toHttp2Headers(lastContent);
+                    final boolean isEmpty = http2Headers == null || http2Headers.isEmpty();
+
+                    if (isEmpty) {
+                        encoder.writeData(ctx, currentStreamId, httpContent.content(), 0, true, promise);
+                        release = false;
+                    } else {
+                        promiseAggregator = new SimpleChannelPromiseAggregator(promise, ctx.channel(), ctx.executor());
+                        encoder.writeData(ctx, currentStreamId, httpContent.content(), 0, false,
+                                promiseAggregator.newPromise());
+                        release = false;
+                        encoder.writeHeaders(ctx, currentStreamId, http2Headers, 0, true,
+                                promiseAggregator.newPromise());
+                        promiseAggregator.doneAllocatingPromises();
+                    }
+
+                    // Reset the stream id
+                    currentStreamId = 0;
```

Comment:
Lets remove this per https://github.com/netty/netty/pull/3975#discussion-diff-34731790